### PR TITLE
feat(specs): create new methods

### PR DESCRIPTION
**Pull request by bengreenbank**
## 🧭 What and Why

I am adding 2 new methods to the Predict API clients:

- Delete a user
- View all users

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/PRED-545?atlOrigin=eyJpIjoiMjdmZmRhZjBiZTg3NGU3YWI3Y2UzZDQ2Y2Q2YzhmOTYiLCJwIjoiaiJ9

### Changes included:

- Create both specs
- Move 'user profile' to common
- Add pagination parameters
- Move some errors to common

## 🧪 Test

Use new endpoints with credentials to test for a response.

### CODE REVIEW

**Review by shortcuts — COMMENTED** [review]
review until https://github.com/algolia/api-clients-automation/pull/975 fixes your build issue

Also, the PR title is used in the changelog, we usually don't put the Jira ticket in it

File: specs/predict/common/schemas/Params.yml [hunk 1]
```diff
@@ -36,3 +44,12 @@ modelsToRetrieve:
 typesToRetrieve:
   type: string
   enum: [properties, segments]
+
+nextPageToken:
+  type: string
+
+previousPageToken:
+  type: string
```
[the remark this answers is not in the document]
Comment (bengreenbank) [reply specs/predict/common/schemas/Params.yml]: Good spot, thanks!

Amended in [840b76f](https://github.com/algolia/api-clients-automation/pull/974/commits/840b76f188794c75a30374da0b86d86330ab17d6)

File: specs/predict/common/schemas/Params.yml [hunk 2]
```diff
@@ -29,10 +37,45 @@ typesToRetrieveParam:
       items:
         $ref: '#/typesToRetrieve'
 
+nextPageTokenParam:
+  type: object
+  required:
+    - nextPageTokenParam
```
Comment (shortcuts) [inline specs/predict/common/schemas/Params.yml:43]: Did you meant to make the `nextPageToken` param required?

Comment (shortcuts) [reply specs/predict/common/schemas/Params.yml:43]: (Same for below occurences)

File: specs/predict/paths/users/fetch.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          $ref: '../../common/schemas/Params.yml#/params'
+  responses:
+    '200':
+      description: OK
+      content:
+        application/json:
+          schema:
+            title: fetchUserProfileResponse
+            required:
+              - user
```
Comment (shortcuts) [inline specs/predict/paths/users/fetch.yml:23]: Here the `ref` might override everything, I believe you need to specify it at the object level directly

**Review by shortcuts — COMMENTED** [review]
Looks great already!

**Review by shortcuts — APPROVED** [review]
Seems good on the API side, do we wait for a review from the Predict team?

**Comment by bengreenbank** [comment]
> Seems good on the API side, do we wait for a review from the Predict team?

@francoischalifour feel free to leave a review if you wish

File: specs/predict/common/schemas/Params.yml [hunk 4]
```diff
@@ -9,6 +9,14 @@ allParams:
     - $ref: '#/modelsToRetrieveParam'
     - $ref: '#/typesToRetrieveParam'
 
+fetchAllUsersParams:
```
Comment (millotp) [inline specs/predict/common/schemas/Params.yml:12]:
```suggestion
fetchAllUserParams:
```
the operationId is called `fetchAllUserProfiles`

File: specs/predict/paths/users/delete.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  responses:
+    '200':
+      description: OK
+      content:
+        application/json:
+          schema:
+            title: deleteUserProfileResponse
+            type: object
+            required:
+              - user
```
Comment (francoischalifour) [inline specs/predict/paths/users/delete.yml:18]: Is `deletedUntil` not always returned?

File: specs/predict/common/parameters.yml [hunk 6]
```diff
@@ -5,3 +5,110 @@ userID:
...
+        computed:
+          type: array
+          description: List of computed segments IDs.
+          items:
+            type: string
+        custom:
+          type: array
+          description: List of custom segments IDs.
+          items:
+            type: string
```
Comment (francoischalifour) [inline specs/predict/common/parameters.yml:114]: Are these properties optional? They're not always returned?

File: specs/predict/common/schemas/Params.yml (hunk 4 above)
Comment (bengreenbank) [reply specs/predict/common/schemas/Params.yml]: [c1cfe7e](https://github.com/algolia/api-clients-automation/pull/974/commits/c1cfe7ee691b3737ac5691a5458761e6fa560536)

Comment (bengreenbank) [reply specs/predict/common/schemas/Params.yml]: I have renamed to match the operationId

File: specs/predict/paths/users/delete.yml (hunk 5 above)
Comment (bengreenbank) [reply specs/predict/paths/users/delete.yml:18]: Correct, I have added in [b57ee9e](https://github.com/algolia/api-clients-automation/pull/974/commits/b57ee9edce63f349bf77e9819b0b56106ef87ad5)

**Review by shortcuts — APPROVED** [review]

File: specs/predict/common/parameters.yml (hunk 6 above)
Comment (bengreenbank) [reply specs/predict/common/parameters.yml:117]: They are both always returned, correct.

Resolved in [1a3ca77](https://github.com/algolia/api-clients-automation/pull/974/commits/1a3ca77142e62eaabd14142edd0949622b09ada5)

**Review by millotp — APPROVED** [review]
Looks good !

**Review by francoischalifour — APPROVED** [review]

**Comment by shortcuts** [comment]
I'll release once the changes have been pushed to main, thanks for the PR!

**Pull request merged**